elasticdl/python/master/master.py

```diff
@@ -430,3 +436,23 @@ def _create_instance_manager(self, args):
             )
 
         return instance_manager
+
+    def _check_timeout_tasks(self):
+        while True:
+            doing_tasks = self.task_d._doing.copy()
+            cur_time = time.time()
+            avg_time = self.master_servicer.get_average_task_complete_time()
+            for task_id, (worker_id, task, start_time) in doing_tasks.items():
+                if task.type in [
+                    elasticdl_pb2.TRAINING,
+                    elasticdl_pb2.EVALUATION,
+                ]:
+                    if (cur_time - start_time) > 3 * avg_time[task.type]:
```

Comment:
If a worker is in the process of a training task, there come some eval tasks, the worker will process these eval tasks first. After the eval tasks are done, the worker will continue the training task. Thus this training task may timeout if using `cur_time - start_time` to determine timeout. If the master has this worker's last report time (report_task_result or report_evaluation_metrics), using `cur_time - worker_last_report_time` will solve this problem.

Reply:
Done. Add an interface `get_worker_liveness_time`